src/webui/www/private/scripts/client.js

```diff
@@ -95,6 +95,61 @@ const getShowFiltersSidebar = function() {
     return (show === null) || (show === 'true');
 };
 
+// v4 and v6 js regex credit https://github.com/sindresorhus/ip-regex/blob/3e220cae3eb66ecfdf4f7678bea7306ceaa41c76/index.js#LL7C1-L25C56
+// license MIT
+function isValidIpAddress(ip) {
+    const v4 = '(?:25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]\\d|\\d)(?:\\.(?:25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]\\d|\\d)){3}';
+    const v6segment = '[a-fA-F\\d]{1,4}';
+
+    const v6 = `(?:(?:${v6segment}:){7}(?:${v6segment}|:)|                                    // 1:2:3:4:5:6:7::  1:2:3:4:5:6:7:8
+    (?:${v6segment}:){6}(?:${v4}|:${v6segment}|:)|                             // 1:2:3:4:5:6::    1:2:3:4:5:6::8   1:2:3:4:5:6::8  1:2:3:4:5:6::1.2.3.4
+    (?:${v6segment}:){5}(?::${v4}|(?::${v6segment}){1,2}|:)|                   // 1:2:3:4:5::      1:2:3:4:5::7:8   1:2:3:4:5::8    1:2:3:4:5::7:1.2.3.4
+    (?:${v6segment}:){4}(?:(?::${v6segment}){0,1}:${v4}|(?::${v6segment}){1,3}|:)| // 1:2:3:4::        1:2:3:4::6:7:8   1:2:3:4::8      1:2:3:4::6:7:1.2.3.4
+    (?:${v6segment}:){3}(?:(?::${v6segment}){0,2}:${v4}|(?::${v6segment}){1,4}|:)| // 1:2:3::          1:2:3::5:6:7:8   1:2:3::8        1:2:3::5:6:7:1.2.3.4
+    (?:${v6segment}:){2}(?:(?::${v6segment}){0,3}:${v4}|(?::${v6segment}){1,5}|:)| // 1:2::            1:2::4:5:6:7:8   1:2::8          1:2::4:5:6:7:1.2.3.4
+    (?:${v6segment}:){1}(?:(?::${v6segment}){0,4}:${v4}|(?::${v6segment}){1,6}|:)| // 1::              1::3:4:5:6:7:8   1::8            1::3:4:5:6:7:1.2.3.4
+    (?::(?:(?::${v6segment}){0,5}:${v4}|(?::${v6segment}){1,7}|:))             // ::2:3:4:5:6:7:8  ::2:3:4:5:6:7:8  ::8             ::1.2.3.4
+    )(?:%[0-9a-zA-Z]{1,})?                                             // %eth0            %1`.
+    replace(/\s*\/\/.*$/gm, '').replace(/\n/g, '').trim();
+    const ipRegex = new RegExp(`(?:^${v4}$)|(?:^${v6}$)`);
+
+    return ipRegex.test(ip);
+}
+
+// getHost emulate the GUI version `QString getHost(const QString &url)`
+function getHost(url) {
+    // We want the domain + tld. Subdomains should be disregarded
+    // If failed to parse the domain or IP address, original input should be returned
+
+    const scheme = url.slice(0, 6).toLowerCase();
+    if (!(scheme.startsWith('http:') || scheme.startsWith('https:') || scheme.startsWith('udp:'))) {
+        return url;
+    }
+
+    try {
+        // hack: URL can not get hostname from udp protocol
+        url = url.replace('udp://', 'https://');
```

Comment:
I would use regex to only match start of the string.
And only match `udp:` without the double slashes.

Reply:
then the correct regex will be even more complicated than the ipv4/ipv6 one.

since the udp one is just the same as http, but the prefix protocol is diff.

for example:

```
udp://tracker.example.com:6969/announce

udp://a.b.c.d:6969/announce
```

using `URL` to handle will save our time, and it is more reliable than regex.

Reply:
I suggest we keep the `hack` for the current now here?